docs/CODE_GUIDELINES.md

```diff
@@ -244,6 +244,24 @@ void Test();
 void Test(void);
 ```
 
+### 3.7. Exceptions to the Formating Rules For Beter Readability
+There are some special situations where vertical alignment and a greater line length does greatly aid readability, for example the initialization of some table-like muiltiple row structures. In these rare cases the layout can be protected from being reformatted when Clang format is applied by adding `// clang-format off` and `// clang-format on` statements either side of the lines of code.
+For example
+```
+// clang-format off
+static const CGUIDialogMediaFilter::Filter filterList[] = {
+  { "movies",       FieldTitle,         556,    SettingType::String,  "edit",   "string",   CDatabaseQueryRule::OPERATOR_CONTAINS },
+  { "movies",       FieldRating,        563,    SettingType::Number,  "range",  "number",   CDatabaseQueryRule::OPERATOR_BETWEEN },
+  { "movies",       FieldUserRating,    38018,  SettingType::Integer, "range",  "integer",  CDatabaseQueryRule::OPERATOR_BETWEEN },
+  ...
+  { "songs",        FieldSource,      39030,    SettingType::List,    "list",   "string",   CDatabaseQueryRule::OPERATOR_EQUALS },
+};  
+// clang-format on
+ ```
+With Clang formatting off the other code guidelines will need to be adhered to manually in the delimited lines of code. Using vertical alignment means that sometimes the entire block of code may need to be realigned, good judgement should be used in each case with the objective of preserving readability yet minimising impact.
```

Comment:
Which guidelines are exclude in `the other guidelines`, because I do not see single exception in this section?

Reply:
Sorry, I don't understand the point you are making @Rechi

Reply:
Having turned the clang check off for a portion of the code, any format the other guidelines e.g. single space after "," need to be applied manually because Clang will not spot them.

Reply:
For me something like, for that special situations the no vertical alignment rule and the maximum line length does not apply, is missing.

Reply:
Read back. The first sentence of the section says "There are some special situations where vertical alignment and a greater line length does greatly aid readability". 

I guess I could repeat and add "and therefore no vertical alignment rule and the maximum line length does not apply". But then again I am also happy for other rules to be relaxed in these rare cases based on human best judgement. I don't want to prejudge which of those rules could be an issue, just give humans the opportunity to use their common sense in _rare_ situations.

But the sentence you are commenting on is simply a reminder of the need for care once the Clang format check is off and I don't think this needs changing.

Reply:
For me that is only an explanation, but not granting any exception.

Reply:
Do we need to state the obvious?
"If the layout produced by applying any of the formatting guidelines in this document is not clearly readable then an exception can be made to any or all of those formatting guidlines." 

I do not understand the desire to nail down every detail, nor do I understand when coding guidlines - style preferences to make code readable and consistent - became rigid rules that must be applied to every line of code regardless of readability.  Sometimes we may need to use human judgement and make exceptions to some or all of the guidelines to get the best readability.